### PR TITLE
Rem 592

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
new page Aging profile , window close on add

### CODE REVIEW

File: src/pages/fi-aging-profile/forms/AgingForm.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+      items: items
+    }
+
+    try {
+      const response = await postRequest({
+        extension: FinancialRepository.AgingProfile.set2,
+        record: JSON.stringify(data)
+      })
+
+      if (!formik.values.recordId) {
```
Comment (omarZrayka23) [inline src/pages/fi-aging-profile/forms/AgingForm.js:103]: you can write if(!obj.recordId)

File: src/pages/fi-aging-profile/forms/AgingForm.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        formik.setFieldValue('recordId', response.recordId)
+        toast.success(platformLabels.Edited)
```
Comment (omarZrayka23) [inline src/pages/fi-aging-profile/forms/AgingForm.js:105]: label is added not Edited

File: src/pages/fi-aging-profile/forms/AgingForm.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+    const data = {
+      header: {
+        recordId: recordId || '',
+        name: obj.name
+      },
+      items: items
+    }
+
+    try {
```
Comment (omarZrayka23) [inline src/pages/fi-aging-profile/forms/AgingForm.js:97]: clear try and catch

**Comment by omarZrayka23** [comment]
Screenshot 2024-10-17 at 8 43 45 AM

error occur when i was fill name and click save

**Comment by omarZrayka23** [comment]
[
Screenshot 2024-10-17 at 8 47 15 AM
](url).   i can not do save

**Comment by imtithalhalalo** [comment]
Screenshot 2024-10-28 161714
transaction log isn't showing records

File: src/pages/fi-aging-profile/forms/AgingForm.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+      extension: FinancialRepository.AgingProfile.set2,
+      record: JSON.stringify(data)
+    })
+
+    if (!obj.recordId) {
+      formik.setFieldValue('recordId', response.recordId)
+      toast.success(platformLabels.Added)
+      window.close()
+    } else {
+      toast.success(platformLabels.Edited)
```
Comment (imtithalhalalo) [inline src/pages/fi-aging-profile/forms/AgingForm.js:107]: do you need also here window.close()?

File: src/pages/fi-aging-profile/forms/AgingForm.js [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+      name: 'caption'
+    }
+  ]
+
+  useEffect(() => {
+    setNumRows(formik.values.agingLeg.length)
+  }, [formik.values.agingLeg])
+
+  function getData() {
+    getRequest({
```
Comment (imtithalhalalo) [inline src/pages/fi-aging-profile/forms/AgingForm.js:131]: no need for .then catch

File: src/pages/fi-aging-profile/index.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      parameters: `_startAt=${_startAt}&_pageSize=${_pageSize}&filter=`
+    })
+
+    return { ...response, _startAt: _startAt }
+  }
+
+  const {
+    query: { data },
+    labels: _labels,
+    access
```
Comment (imtithalhalalo) [inline src/pages/fi-aging-profile/index.js:36]: use invalidate

**Comment by imtithalhalalo** [comment]
Screenshot 2024-10-28 163843
when i save with empty rows

File: src/pages/fi-aging-profile/forms/AgingForm.js (hunk 4 above)
Comment (Alinadimmokdad) [reply src/pages/fi-aging-profile/forms/AgingForm.js:107]: just on Add

**Review by imtithalhalalo — APPROVED** [review]

**Pull request merged**